### ISSUE TITLE
use_build_context_synchronously triggered incorrectly in if statement

### ISSUE THREAD

**Issue by hyungtaecf** [dart-lang/linter#4027]
**Describe the issue**
I believe use_build_context_synchronously is triggering incorrectly in this code. I spent some time analyzing it and there are no async gaps before the part `!await _tryToPurchaseSong(context)`, so the lint shouldn't have triggered it.

If I put that line before the if statement then the linter doesn't say anything but If I put inside the if statement, even as a single condition then the linter triggers.

**To Reproduce**
``` dart
Future<void> onTapSong({required BuildContext context}) async {
   final appKey = getAppKey(context);
    if (comingSoon && !hasRole(StaffRole.songManager, context: context)) return;
    final previewSongScreen = FeatureFlag.previewSongScreen.getValue(context);
    if (!previewSongScreen &&
        !isLibrarySong &&
        // ignore: use_build_context_synchronously
        !await _tryToPurchaseSong(context)) return;
    if (appKey.currentContext == null) return;
    goToSongScreen(
      appKey.currentContext!,
      songId: song.id,
      imageUrl: song.imageUrl,
      artistAvatarHeroTag: _artistAvatarHeroTag,
      videoHeroTag: _videoHeroTag,
      songInfoHeroTag: _songInfoHeroTag,
      comingFromHomeScreen: comingFromHomeScreen,
    );
}
```

**Expected behavior**
The linter should not trigger use_build_context_synchronously if there are no async gaps.

**Comment by slavap** [comment]
More simple case:
```dart
client.clientDocsVal = newDocs;
if (!await _editClientStore.updateClient(context, client)) client.clientDocsVal = docs;
```
If changed like that, no warning:
```dart
var updRslt = await _editClientStore.updateClient(context, client);
if (!updRslt) client.clientDocsVal = docs;
```
